crates/virtio-queue/src/lib.rs

```diff
@@ -85,6 +88,16 @@ pub struct Descriptor {
 
 #[allow(clippy::len_without_is_empty)]
 impl Descriptor {
+    /// Creates a new descriptor
+    pub fn new(addr: u64, len: u32, flags: u16, next: u16) -> Self {
```

Comment:
This should only be enabled for tests and when the `test-utils` feature is used.